### PR TITLE
[DEPRECATION] (WIP) calling `this._super` when there is no method on parent class

### PR DESCRIPTION
**Pull request by givanse**
https://github.com/emberjs/ember.js/issues/14780

 - [ ] document  `didDestroyElement`, only available in interactive mode

### CODE REVIEW

File: packages/ember-metal/lib/mixin.js [hunk 1]
```diff
@@ -129,6 +138,23 @@ function giveMethodSuper(obj, key, method, values, descs) {
...
+    /*
+    deprecate(
+      'Calling `_super` when there is no method on the parent class is deprecated.',
+      false,
+      { id: 'ember-metal.missing-super-computed', until: '3.0.0' }
+    );
+    *
+    */
+    /*
+    deprecate(
```
Comment (givanse) [inline packages/ember-metal/lib/mixin.js]: copy-pasta from within this file as a sanity check, it errors with:
```
Uncaught Error: Ember.required is deprecated as its behavior is inconsistent and unreliable.
Uncaught TypeError: Cannot read property 'registerDeprecationHandler' of undefined
Uncaught TypeError: Cannot read property 'extend' of undefined
```

File: packages/ember-metal/lib/mixin.js [hunk 2]
```diff
@@ -129,6 +138,23 @@ function giveMethodSuper(obj, key, method, values, descs) {
 
   // Only wrap the new method if the original method was a function
   if (superMethod === undefined || 'function' !== typeof superMethod) {
+    /*
+    deprecate(
```
Comment (givanse) [inline packages/ember-metal/lib/mixin.js]:
```
Uncaught Error: Calling `_super` when there is no method on the parent class is deprecated.
```

:thinking:

Comment (givanse) [inline packages/ember-metal/lib/mixin.js]: `deprecate` is meant to be executed at app runtime

**Comment by givanse** [comment]
The last broken test is:

> Ember.Mixin#reopen: using reopen() and calling _super where there is not a super function does not cause infinite recursion

Previous fixes are found in these PRs https://github.com/emberjs/ember.js/pull/9678 https://github.com/emberjs/ember.js/pull/12062 I'm still trying to piece together what is going on with that code, any hints are appreciated.

**Comment by locks** [comment]
Apologies for letting this work linger all this time 😞  At this point I think rebasing this PR would be more work than starting anew so I'm closing it for now.

**Pull request closed without merging**